pkg/etcd/etcd.go

```diff
@@ -1248,8 +1249,17 @@ func (e *ETCD) setEtcdStatusCondition(node *v1.Node, client kubernetes.Interface
 	}
 
 	if find, condition := nodeUtil.GetNodeCondition(&node.Status, etcdStatusType); find >= 0 {
-		if condition.Status == newCondition.Status && memberStatus != StatusUnjoined {
+
+		// if the condition is not changing, we only want to update the last heartbeat time
+		if condition.Status == newCondition.Status && condition.Reason == newCondition.Reason && condition.Message == newCondition.Message {
 			logrus.Debugf("Node %s is not changing etcd status condition", memberName)
+
+			// If the condition status is not changing, we only want to update the last heartbeat time if the
+			// LastHeartbeatTime is older than the heartbeatTimeout.
+			if metav1.Now().Sub(condition.LastHeartbeatTime.Time) >= heartbeatTimeout {
```

Comment:
I think this is backwards? We want to return early and skip the node condition update if the time since the last heartbeat is LESS than the timeout, not greater.

Also, calling this a timeout probably isn't correct, as it is an interval. You might update the var name.

```suggestion
			if metav1.Now().Sub(condition.LastHeartbeatTime.Time) < heartbeatTimeout {
```